primitives/trie/src/storage_proof.rs

```diff
@@ -31,6 +31,12 @@ pub struct StorageProof {
 	trie_nodes: Vec<Vec<u8>>,
 }
 
+/// Storage proof in compact form.
+#[derive(Debug, PartialEq, Eq, Clone, Encode, Decode)]
+pub struct CompactProof {
```

Comment:
Why don't we "compress" by default? Why do we need a second representation for this?

Reply:
The PR just expose the trie functionality, I did not want to enforce compression, so at this point the main target was only cumulus, not light client for instance.

Reply:
But wouldn't this also be helpful for light clients?

Reply:
Replies to light client contains storage proof, so they can be compressed as well.

Reply:
Sorry, I misread your question. So yes I think it could be helpful, but for the pov it seems more obvious to me that it is useful.
For light client we can have a lot more single value query or small proof where it will not really shine and maybe the computational cost of the compaction may not be good, so I was starting to propose this for the best use case.